src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1431,6 +1431,49 @@ function nonunique(df::AbstractDataFrame, cols)
     end
 end
 
+"""
+    allunique(df::AbstractDataFrame, cols=:)
+
+Return `true` if all rows of `df` are not duplicated. Two rows are duplicate if
+all their columns contain equal values (according to `isequal`).
+
+See also [`unique`](@ref) and [`nonunique`](@ref).
+
+# Arguments
+- `df` : `AbstractDataFrame`
+- `cols` : a selector specifying the column(s) or their transformations to compare.
+  Can be any column selector or transformation accepted by [`select`](@ref).
+
+# Examples
+
+```jldoctest
+julia> df = DataFrame(i=1:4, x=[1, 2, 1, 2])
+4×2 DataFrame
+ Row │ i      x
+     │ Int64  Int64
+─────┼──────────────
+   1 │     1      1
+   2 │     2      2
+   3 │     3      1
+   4 │     4      2
+
+julia> allunique(df)
+true
+
+julia> allunique(df, :x)
+false
+
+julia> allunique(df, :i => ByRow(isodd))
+false
+```
+"""
+function Base.allunique(df::AbstractDataFrame, cols=:)
+    udf = select(df, cols, copycols=false)
```

Comment:
This will throw an error if a transformation is passed and `df` is a `SubDataFrame`, right?

Reply:
yes, after #3231 (as I noticed some cases that we allowed to slip through). For `SubDataFrame` only column selection is allowed.

I have this tested (and that is why tests fail currently).

Reply:
I mean, maybe we should pass `copycols=!df isa SubDataFrame` so that transformations can be used even with `SubDataFrame`?

Reply:
A good point. Changed. The same applied to `describe` and `nonunique`.